iron-overlay-manager.html

```diff
@@ -13,199 +13,261 @@
 <script>
 
   /**
-   * @struct
-   * @constructor
+   * Singleton instance.
+   * This is compiler-friendly, because properties can be safely renamed.
+   * @typedef {Object} Polymer.IronOverlayManager
```

Comment:
Huh, I didn't realize this would work. This does type checking? If you misspell a property of IronOverlayManager, or treat a property as the wrong type, does the compiler throw an error?

If not, that's great!